### ISSUE TITLE
0.5.0 keep crash

### ISSUE THREAD

**Issue by svetoslav0625** [StasDoskalenko/react-native-google-fit#43]
android app keep crash on loading without any error message.

react-native-google-fit: 0.5.0
react-native: 0.44.0

**Comment by StasDoskalenko** [comment]
@svetoslav0625 could you please provide your code and some android studio stack trace

**Comment by svetoslav0625** [comment]
import GoogleFit from 'react-native-google-fit'

      if (platform === 'ios') {
        // use Apple Health Kit
        ...
      } else {
        // use Google fit
        GoogleFit.authorize()

        // ...
        // Call when authorized
        this.requestAndroidFineLocation().then(() => {
          if (this.locationPermission) {
            this.getGoogleFitDataWithPermission()
          } else {
            this.getGoogleFitDataWithoutPermission()
          }
          GoogleFit.observeSteps(evt => {
            this.requestAndroidFineLocation().then(() => {
              if (this.locationPermission) {
                this.getGoogleFitDataWithPermission()
              } else {
                this.getGoogleFitDataWithoutPermission()
              }
            })
          })
        })
        // GoogleFit.startRecording(callback => {
        //   // Process data from Google Fit Recording API (no google fit app needed)
        // })
      }

**Comment by StasDoskalenko** [comment]
@svetoslav0625 you should use ```GoogleFit.onAuthorize(() => {
  some callback or promise or Redux action to work
});```
to be sure whether your app is authorized

**Comment by svetoslav0625** [comment]
0.5.0 still crash immediately after launch.
I upgraded version to 0.5.2 then crash has gone.
But after upgraded, GoogleFit.isAvailable always returns false.

**Comment by svetoslav0625** [comment]
@StasDoskalenko, GoogleFit.onAuthorize() is never called.
Is there any update?

**Comment by StasDoskalenko** [comment]
@svetoslav0625 what message do you get in
```GoogleFit.onAuthorizeFailure((res) => {
  console.log(res)
});```

**Comment by svetoslav0625** [comment]
`GoogleFit.onAuthorizeFailure()` is never called. `GoogleFit.isAvailable()` always returns false. @StasDoskalenko

**Comment by StasDoskalenko** [comment]
@svetoslav0625 firstly
try to remove node_modules, and reinstall modules.
Then, try to clear gradle cache.
And does this code work for you?
```javascript
componentDidMount() {
      GoogleFit.onAuthorize((response) => {
        console.log('response >>>', response)
        GoogleFit.startRecording((callback) => {
          console.log('callback >>>', callback) // u should call startRecording only after authorize
        })
      })

      GoogleFit.onAuthorizeFailure((res) => {
        console.log('response fail >>> ', res)
      })

      GoogleFit.authorize()
    }
```

**Comment by svetoslav0625** [comment]
I tried to remove node_modules and reinstall package, clear gradle and build cache.
But `GoogleFit.authorize()` is still not working...

**Comment by svetoslav0625** [comment]
what is your recommended version? @StasDoskalenko

**Comment by StasDoskalenko** [comment]
@svetoslav0625 it might be because of too old react-native.

**Comment by svetoslav0625** [comment]
I found problem. I am using third party library and it is using different google play service version.
so after recent upgrade of google play service, there is some conflict with react-native google fit.
I added this line to my build.gradle and it works fine. @StasDoskalenko 

  ` compile 'com.google.android.gms:play-services-auth:12.0.1'`
  ` compile 'com.google.android.gms:play-services-fitness:12.0.1'`
   `compile 'com.google.android.gms:play-services-gcm:12.0.1'`
